debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/connection/wal2json/NonStreamingWal2JsonMessageDecoder.java

```diff
@@ -53,7 +55,7 @@ public void processMessage(ByteBuffer buffer, ReplicationMessageProcessor proces
             LOGGER.debug("Message arrived for decoding {}", message);
             final long txId = message.getLong("xid");
             final String timestamp = message.getString("timestamp");
-            final long commitTime = dateTime.systemTimestamp(timestamp);
+            final Instant commitTime = Conversions.toInstant(dateTime.systemTimestamp(timestamp));
```

Comment:
@gunnarmorling Should not `systemTimestamp` return `Instant` instead of doing conversion here?

Reply:
Ideally yes. But I was trying, this is like pulling Sphaghetti from the plate, never ending. So I decided to stop here, and leave that for another time.

Reply:
Fair enough :-)